### DISCUSSION TITLE
Why the RGCNConv does not support x_dict and edge_index_dict?

### DISCUSSION THREAD

**Discussion by ver228** [pyg-team/pytorch_geometric#8263] [General]
I have been using the heterogenous graph functionalities of PyG and they work great. Thanks for developing them!
 
However, now I would like to bench mark against a R-GCN model, and realise that the `RGCNConv` layer needs a flatten version of the graph, in contrast from the `HeteroConv` that can receive the` HeteroData` `x_dict` and `edge_index_dict`. 

Is there a hard reason for this or it just that this module was developed before the HeteroData format was developed?

**Comment by rusty1s** [comment]
`R-GCN` is designed to only be applicable on equally sized feature matrices. However, `to_hetero` and R-GCN are actually equivalent given the `GraphConv`/`SAGEConv` operator, see https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/test_to_hetero_transformer.py#L384.

**Comment by ver228** [comment]
Thanks! That is a very nice little example, you should consider to add it the tutorial or at least to the examples folder :)
